packit/schema.py

```diff
@@ -167,26 +167,59 @@ def make_instance(self, data, **kwargs):
         return NotificationsConfig(**data)
 
 
-class TargetField(fields.String):
-    def _deserialize(
-        self,
-        target,
-        *args,
-        **kwargs,
-    ):
-        if target in DEPRECATED_TARGET_MAP:
-            logger.warning(
-                f"Target '{target}' is deprecated. Please update your configuration "
-                f"file and use '{DEPRECATED_TARGET_MAP[target]}' instead."
+class TargetsListOrDict(fields.Field):
+    """Field type expecting a List[str] or Dict[str, Dict[str, Any]]
+    Union is not supported by marshmallow so we have to validate manually :(
+    https://github.com/marshmallow-code/marshmallow/issues/1191
+    """
+
+    @staticmethod
+    def __is_targets_dict(value) -> bool:
+        if (
+            not isinstance(value, dict)
+            or not all(isinstance(k, str) for k in value.keys())
+            or not all(isinstance(v, dict) for v in value.values())
+        ):
+            return False
+        # check the 'attributes', e.g. {'distros': ['centos-7']}
+        for attr in value.values():
+            for key, value in attr.items():
+                if key != "distros":
+                    raise ValidationError(f"Unknown key {key!r} in {attr!r}")
```

Comment:
Does it make sense to use a dictionary if there is only one valid key?

What about using  `List[str]` or `Dict[str,List[str]]`?

Reply:
On the other hand, the example you mentioned in the comment looks good from the user perspective (If they know what does the term `distro` mean here).

Reply:
There's only one now, yes, so you're right.
But who knows, there might be more later since the `targets` metadata is used in all jobs, other jobs might want to add some "attributes" to the targets as well. And I don't want to do another "list or dict" (for backward compatibility) later for the targets.
I even think it's more descriptive to have
```yaml
- job: tests
  metadata:
    targets:
      epel-7-x86_64:
        distros: [centos-7]
```
instead of 
```yaml
- job: tests
  metadata:
    targets:
      epel-7-x86_64: [centos-7]
```

Reply:
I'm only not sure if people know what does `distro` mean here. If yes, this is definitely more descriptive.

Reply:
`distro` is the [tmt term](https://tmt.readthedocs.io/en/latest/spec/context.html). We might want to elaborate something more descriptive, like e.g. `test_on_distros`, but I think as long as it's described in the documentation, the `distro` is fine.

Reply:
`test_on_distro` looks bettter to me, but I am ok with `distro` as well. Thanks for the clarification.